lib/wibox/widget/slider.lua

```diff
@@ -0,0 +1,548 @@
+            cr:restore()
+        else
+            cr:stroke()
+        end
+    end
+
+    cr:translate(-x_offset, -y_offset)
+
+    -- Paint the handle
+    local handle_color = self._private.handle_color
+        or beautiful.slider_handle_color
+
+    -- It is ok if there is no color, it will be inherited
+    if handle_color then
+        cr:set_source(color(handle_color))
+    end
+
+    local handle_height, handle_width = height, self._private.handle_width
+        or beautiful.slider_handle_width
+        or height/2
+
+    local handle_shape = self._private.handle_shape
+        or beautiful.slider_handle_shape
+        or properties.handle_shape
+
+    -- Lets get the margins for the handle
+    local margins = self._private.handle_margins
+        or beautiful.slider_handle_margins
+
+    local x_offset, y_offset = 0, 0
+
+    if margins then
+        if type(margins) == "number" then
+            x_offset, y_offset = margins, margins
+            handle_width  = handle_width  - 2*margins
+            handle_height = handle_height - 2*margins
+        else
+            x_offset, y_offset = margins.left or 0, margins.top or 0
+            handle_width  = handle_width  -
+                (margins.left or 0) - (margins.right  or 0)
+            handle_height = handle_height -
+                (margins.top  or 0) - (margins.bottom or 0)
+        end
+    end
+
+    local value = self._private.value or self._private.min or 0
+
+    -- Get the widget size back to it's non-transfored value
+    --local matrix = gmatrix.from_cairo_matrix(cr:get_matrix())
+    local min, _, interval = get_extremums(self)
+    local rel_value = ((value-min)/interval) * (width-handle_width)
+
+    cr:translate(x_offset + rel_value, y_offset)
+
+    local handle_border_width = self._private.handle_border_width
+        or beautiful.slider_handle_border_width
+        or properties.handle_border_width or 0
+
+    handle_shape(cr, handle_width, handle_height)
+
+    if handle_border_width > 0 then
+        cr:fill_preserve()
+    else
+        cr:fill()
+    end
+
+    -- Draw the handle border
+    if handle_border_width > 0 then
+        local handle_border_color = self._private.handle_border_color
+            or beautiful.slider_handle_border_color
+            or properties.handle_border_color
+
+        if handle_border_color then
+            cr:set_source(color(handle_border_color))
+        end
+
+        cr:set_line_width(handle_border_width)
+        cr:stroke()
+    end
+end
+
+function slider:fit(_, width, height)
+    -- Use all the space, this should be used with a constraint widget
+    return width, height
+end
+
+local event = {}
+
+function event.mouse_enter(self)
+    self:emit_signal("widget::redraw_needed") --TODO
+    local cursor = self._private.cursor or "fleur"
+    capi.root.cursor(cursor)
+end
+
+function event.mouse_leave(self)
+    self:emit_signal("widget::redraw_needed") --TODO
+--     capi.root.cursor("fleur")
```

Comment:
This seems unfinished... The cursor is changed, but never reset? Also, why is the cursor on the root window changed? That seems useless and not related to this widget at all. (And no, right now I have no good idea nor much interest to implement per-widget mouse cursors)